astroid/constraint.py

```diff
@@ -0,0 +1,137 @@
+# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
```

Comment:
Just thinking in terms of future expandability:
Should we make this a module with a `__init__.py` and put `get_constraints` in a `__all__`.

I can see these classes and functions change over time as we might expand the control flow inference and by putting things explicitly in `__all__` we don't need to worry about backwards incompatibility.